### ISSUE TITLE
Analog for `strict-boolean-expressions` typescript-eslint rule

### ISSUE THREAD

**Issue by basil-gor** [vuejs/eslint-plugin-vue#2143]
**Please describe what the rule should do:**

There are very useful rule [strict-boolean-expressions](https://github.com/typescript-eslint/typescript-eslint/blob/4f6be7f3ec46e673dcdab29d67a3f53957a8ecdd/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md) in [typescript-eslint](https://github.com/typescript-eslint).
If there are no possible ways to use it in vue templates, maybe we can implement a vue analog for this rule?

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

_Just for example for string expression:_
```vue
// Will be incorrect ❌
<div
  v-if="someEmptyString"
/>

// Will be correct ✅
<div
   v-if="someEmptyString !== ''"
/>
```

**Additional context**

**Comment by juliusfriedman** [comment]
Have been wondering if the fact `@typescript-eslint/strict-boolean-expressions` is only working inside `<script>` tags and not templates is a bug.

It comes as a surprise to me that `v-if` parsing and `if` parsing are not handled the same, I would think they are the same syntax nodes in AST and that this rule would work out of the box.

**Comment by waynzh** [comment]
> if the fact `@typescript-eslint/strict-boolean-expressions` is only working inside `<script>` tags

Exactly. Since the Node types in the `<template>` are different, similar functionality needs to be implemented to make it work.
